tests/Core/Tokenizer/BackfillFnTokenTest.php

```diff
@@ -553,27 +531,85 @@ public function testTernary()
 
 
     /**
-     * Test that the backfill presumes T_FN during live coding, but doesn't set the additional index keys.
+     * Test arrow function nested within a method declaration.
      *
      * @covers PHP_CodeSniffer\Tokenizers\PHP::processAdditional
      *
      * @return void
      */
-    public function testLiveCoding()
+    public function testNestedInMethod()
+    {
+        $tokens = self::$phpcsFile->getTokens();
+
+        $token = $this->getTargetToken('/* testNestedInMethod */', T_FN);
+        $this->backfillHelper($token);
+
+        $this->assertSame($tokens[$token]['scope_opener'], ($token + 5), 'Scope opener is not the arrow token');
+        $this->assertSame($tokens[$token]['scope_closer'], ($token + 17), 'Scope closer is not the semicolon token');
+
+        $opener = $tokens[$token]['scope_opener'];
+        $this->assertSame($tokens[$opener]['scope_opener'], ($token + 5), 'Opener scope opener is not the arrow token');
+        $this->assertSame($tokens[$opener]['scope_closer'], ($token + 17), 'Opener scope closer is not the semicolon token');
+
+        $closer = $tokens[$token]['scope_opener'];
+        $this->assertSame($tokens[$closer]['scope_opener'], ($token + 5), 'Closer scope opener is not the arrow token');
+        $this->assertSame($tokens[$closer]['scope_closer'], ($token + 17), 'Closer scope closer is not the semicolon token');
+
+    }//end testNestedInMethod()
+
+
+    /**
+     * Verify that "fn" keywords which are not arrow functions get tokenized as T_STRING and don't
+     * have the extra token array indexes.
+     *
+     * @param string $testMarker The comment prefacing the target token.
+     *
+     * @dataProvider dataNotAnArrowFunction
+     * @covers       PHP_CodeSniffer\Tokenizers\PHP::processAdditional
+     *
+     * @return void
+     */
+    public function testNotAnArrowFunction($testMarker)
     {
         $tokens = self::$phpcsFile->getTokens();
 
-        $token = $this->getTargetToken('/* testLiveCoding */', [T_STRING, T_FN]);
-        $this->assertSame($tokens[$token]['code'], T_FN, 'Token not tokenized as T_FN');
+        $token      = $this->getTargetToken('/* testFunctionName */', [T_STRING, T_FN], 'fn');
```

Comment:
The test marker is hard-coded here so it's not actually using the data provider.

Reply:
Thanks for catching that. Working on too many things at the same time :cold_sweat:

Reply:
Resolved via #2863